### PR TITLE
Add Falco enforcement policy

### PR DESCRIPTION
**Pull request by airadier**
Enforce the installation of Falco Operator and a Falco instance in all clusters tagged "falco: true"

### CODE REVIEW

**Comment by airadier** [comment]
Hi @ycao56 please review latest commit, updating to latest apiVersion

File: community/SI-System-and-Information-Integrity/policy-falco.yaml [hunk 1]
```diff
@@ -0,0 +1,384 @@
+apiVersion: policy.open-cluster-management.io/v1
+kind: Policy
+metadata:
+  name: policy-falco
+  namespace: falco-manager
```
Comment (ycao56) [inline community/SI-System-and-Information-Integrity/policy-falco.yaml]: remove namespace

File: community/SI-System-and-Information-Integrity/policy-falco.yaml [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+  annotations:
+    policy.open-cluster-management.io/standards: NIST SP 800-53
+    policy.open-cluster-management.io/categories: SI System and Information Integrity
+    policy.open-cluster-management.io/controls: SI-5 Security Alerts, Advisories, and Directives
+spec:
+  complianceType: musthave
+  remediationAction: enforce
+  disabled: false
+  namespaces:
```
Comment (ycao56) [inline community/SI-System-and-Information-Integrity/policy-falco.yaml]: remove this field. it is no longer needed

File: community/SI-System-and-Information-Integrity/policy-falco.yaml [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+kind: Policy
+metadata:
+  name: policy-falco
+  namespace: falco-manager
+  annotations:
+    policy.open-cluster-management.io/standards: NIST SP 800-53
+    policy.open-cluster-management.io/categories: SI System and Information Integrity
+    policy.open-cluster-management.io/controls: SI-5 Security Alerts, Advisories, and Directives
+spec:
+  complianceType: musthave
```
Comment (ycao56) [inline community/SI-System-and-Information-Integrity/policy-falco.yaml]: remove this field. it is no longer needed

File: community/SI-System-and-Information-Integrity/policy-falco.yaml [hunk 4]
```diff
@@ -0,0 +1,384 @@
...
+  remediationAction: enforce
+  disabled: false
+  namespaces:
+    exclude: ["kube-*"]
+    include: ["*"]
+
```
Comment (ycao56) [inline community/SI-System-and-Information-Integrity/policy-falco.yaml]: remove empty line

File: community/SI-System-and-Information-Integrity/policy-falco.yaml [hunk 5]
```diff
@@ -0,0 +1,384 @@
...
+          namespaceSelector:
+            exclude: ["kube-*"]
+            include: ["*"]
+          object-templates:
+            - complianceType: musthave
+              objectDefinition:
+                apiVersion: policy.open-cluster-management.io/v1
+                kind: ConfigurationPolicy
+                metadata:
+                  name: policy-falco-operatorgroup
```
Comment (ycao56) [inline community/SI-System-and-Information-Integrity/policy-falco.yaml]: there is already a `ConfigurationPolicy` defined on line 280. This part is redundant

File: community/SI-System-and-Information-Integrity/policy-falco.yaml [hunk 6]
```diff
@@ -0,0 +1,384 @@
...
+subjects:
+- name: policy-falco
+  kind: Policy
+  apiGroup: policy.open-cluster-management.io
+---
+apiVersion: apps.open-cluster-management.io/v1
+kind: PlacementRule
+metadata:
+  name: placement-policy-falco
+  namespace: falco-manager
```
Comment (ycao56) [inline community/SI-System-and-Information-Integrity/policy-falco.yaml]: remove namespace

File: community/SI-System-and-Information-Integrity/policy-falco.yaml [hunk 7]
```diff
@@ -0,0 +1,384 @@
...
+                metadata:
+                  name: falco-enforcement
+                  namespace: falco-operator
+                spec: {}
+---
+apiVersion: policy.open-cluster-management.io/v1
+kind: PlacementBinding
+metadata:
+  name: binding-policy-falco
+  namespace: falco-manager
```
Comment (ycao56) [inline community/SI-System-and-Information-Integrity/policy-falco.yaml]: remove namespace

File: community/SI-System-and-Information-Integrity/policy-falco.yaml [hunk 8]
```diff
@@ -0,0 +1,384 @@
...
+kind: PlacementRule
+metadata:
+  name: placement-policy-falco
+  namespace: falco-manager
+spec:
+  clusterConditions:
+  - type: OK
+  clusterSelector:
+    matchLabels:
+      falco: "true"
```
Comment (ycao56) [inline community/SI-System-and-Information-Integrity/policy-falco.yaml]: any special reason to use this cluster selector? if not, I recommend change it to 
```
  clusterSelector:
    matchExpressions:
      - {key: environment, operator: In, values: ["dev"]}
```

File: community/SI-System-and-Information-Integrity/policy-falco.yaml [hunk 9]
```diff
@@ -0,0 +1,384 @@
...
+  kind: Policy
+  apiGroup: policy.open-cluster-management.io
+---
+apiVersion: apps.open-cluster-management.io/v1
+kind: PlacementRule
+metadata:
+  name: placement-policy-falco
+  namespace: falco-manager
+spec:
+  clusterConditions:
```
Comment (ycao56) [inline community/SI-System-and-Information-Integrity/policy-falco.yaml:355]: should be 
```
  clusterConditions:
  - status: "True"
    type: ManagedClusterConditionAvailable
```

File: community/SI-System-and-Information-Integrity/policy-falco.yaml [hunk 10]
```diff
@@ -0,0 +1,384 @@
...
+                      objectDefinition:
+                        apiVersion: operators.coreos.com/v1
+                        kind: OperatorGroup
+                        metadata:
+                          name: falco-operatorgroup
+                          namespace: falco-operator
+                        spec:
+                          targetNamespaces:
+                          - falco-operator
+
```
Comment (ycao56) [inline community/SI-System-and-Information-Integrity/policy-falco.yaml]: remove extra line

File: community/SI-System-and-Information-Integrity/policy-falco.yaml [hunk 11]
```diff
@@ -0,0 +1,384 @@
...
+                  name: falco-operator
+                  namespace: falco-operator
+                spec:
+                  channel: stable
+                  installPlanApproval: Automatic
+                  name: falco-certified
+                  source: certified-operators
+                  sourceNamespace: openshift-marketplace
+                  startingCSV: falco-operator.v1.2.0
+
```
Comment (ycao56) [inline community/SI-System-and-Information-Integrity/policy-falco.yaml]: remove extra line

File: community/SI-System-and-Information-Integrity/policy-falco.yaml [hunk 12]
```diff
@@ -0,0 +1,384 @@
...
+                  installModes:
+                    - type: OwnNamespace
+                      supported: true
+                    - type: SingleNamespace
+                      supported: true
+                    - type: MultiNamespace
+                      supported: false
+                    - type: AllNamespaces
+                      supported: false
+
```
Comment (ycao56) [inline community/SI-System-and-Information-Integrity/policy-falco.yaml]: remove extra line

**Comment by ycao56** [comment]
Not required but I recommend removing 
```
          namespaceSelector:
            exclude: ["kube-*"]
            include: ["*"]
```
in every configuration policy as you already hardcoded namespace in the objectTemplate. It has higher priority thus `namespaceSelector` would be ignored.

File: community/SI-System-and-Information-Integrity/policy-falco.yaml (hunk 8 above)
Comment (airadier) [reply community/SI-System-and-Information-Integrity/policy-falco.yaml]: Not really, in our tests we are using the clusters labeled with "falco: true", but I can see this selector is used in all policy examples, so I am changing it.

File: community/SI-System-and-Information-Integrity/policy-falco.yaml (hunk 5 above)
Comment (airadier) [reply community/SI-System-and-Information-Integrity/policy-falco.yaml]: Ups, probably a copy-paste and indentation error! Fixing.

**Review by ycao56 — APPROVED** [review]

**Pull request merged**